.github/workflows/publish.yml

```diff
@@ -14,10 +14,22 @@ jobs:
           - [self-hosted, macos, arm64]
     runs-on: ${{ matrix.os }}
     steps:
+      - name: Set up Homebrew self-hosted
+        if: contains(${{ matrix.os }}, 'self-hosted')
+        env:
+          # This environment variable is not documented, but is needed to
+          # prevent failures and destructive actions added in this commit:
+          # https://github.com/Homebrew/actions/commit/616a563beaa0715ddebbb39a331224af1b46fdaa
+          GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED: 1
+        uses: Homebrew/actions/setup-homebrew@master
+        with:
+          debug: true
+
       - name: Set up Homebrew
-        # NOTE: A specific commit is used here b/c this action does not
-        #       publish tags and the current `master` appears to be broken.
-        uses: Homebrew/actions/setup-homebrew@449449e64aaa5faac0d24229a2ffba32438e4b63
+        if: contains(${{ matrix.os }}, 'self-hosted') == false
+        uses: Homebrew/actions/setup-homebrew@master
+        with:
+          debug: true
```

Comment:
Is there a way we can do this without splitting the step into two conditional steps?

Something like this:
```yaml
  build-bottles:
    strategy:
      matrix:
        os:
          - ubuntu-20.04
          - macos-latest
        self_hosted: ''
        include:
          - os: [self-hosted, macos, arm64]
            self_hosted: '1'
    runs-on: ${{ matrix.os }}
    steps:
      - name: Set up Homebrew
        env:
          # This environment variable is not documented, but is needed to
          # prevent failures and destructive actions added in this commit:
          # https://github.com/Homebrew/actions/commit/616a563beaa0715ddebbb39a331224af1b46fdaa
          GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED: ${{ matrix.self_hosted }}
        uses: Homebrew/actions/setup-homebrew@master
        with:
          debug: true
```

That way we avoid repeating the `uses` and `with` keys, etc.

Reply:
That looks like it could work...lets try it!

Reply:
@kylewillmon and I talked offline and came to a different conclusion about the best way to stay DRY.